### PR TITLE
fix: ensure styles are loaded in dev mode for routes containing special characters

### PR DESCRIPTION
**Pull request by whataboutpereira**
SvelteKit doesn't decode special characters in pathnames when loading CSS modules in dev mode, resulting in an error:

```
Internal server error: Failed to load url /src/routes/(special)/hinnap%C3%A4ring/+page.svelte?svelte=&type=style&lang.css=&inline= (resolved id: /src/routes/(special)/hinnap%C3%A4ring/+page.svelte?svelte&type=style&lang.css). Does the file exist?
```

Actual path:

`/src/routes/(special)/hinnapäring/`

Fix by using decodeURI on the url.pathname when loading CSS modules.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/src/exports/vite/dev/index.js [hunk 1]
```diff
@@ -187,9 +188,7 @@ export async function dev(vite, vite_config, svelte_config) {
 									query.set('inline', '');
 
 									try {
-										const mod = await loud_ssr_load_module(
-											`${url.pathname}${url.search}${url.hash}`
-										);
+										const mod = await loud_ssr_load_module(`${pathname}${url.search}${url.hash}`);
```
Comment (benmccann) [inline packages/kit/src/exports/vite/dev/index.js:191]: by doing the `decodeURI` call outside the `if` you're doing it all the time, but really you only need to do it for CSS requests
```suggestion
										const mod = await loud_ssr_load_module(`${decodeURI(url.pathname)}${url.search}${url.hash}`);
```

**Comment by Rich-Harris** [comment]
Thank you. This would benefit from a test, are you able to add one?

**Comment by whataboutpereira** [comment]
> Thank you. This would benefit from a test, are you able to add one?

I did think about that, but unfortunately I wouldn't know where to begin.

**Comment by whataboutpereira** [comment]
> I added a test in [cea820e](https://github.com/sveltejs/kit/commit/cea820e8f49869bb6f58811941a4897b1c46ac4f)

Wonderful, thank you!

**Pull request merged**